command/alloc_logs.go

```diff
@@ -186,22 +192,25 @@ func (l *AllocLogsCommand) Run(args []string) int {
 		return 1
 	}
 
-	var task string
-	if len(args) >= 2 {
-		task = args[1]
-		if task == "" {
-			l.Ui.Error("Task name required")
-			return 1
-		}
-
+	// if -task isn't provided fallback to reading the task name
+	// from args.
+	if task != "" {
+		err = validateTaskExistsInAllocation(task, alloc)
 	} else {
-		task, err = lookupAllocTask(alloc)
-
-		if err != nil {
-			l.Ui.Error(err.Error())
-			return 1
+		if len(args) >= 2 {
+			task = args[1]
+			if task == "" {
+				l.Ui.Error("Task name required")
+				return 1
+			}
+		} else {
+			task, err = lookupAllocTask(alloc)
 		}
 	}
+	if err != nil {
+		l.Ui.Error(err.Error())
```

Comment:
Some context around the potential error here could be nice
```suggestion
		l.Ui.Error(fmt.Sprintf("Failed to validate task", err.Error())
```